### PR TITLE
Modern Web Apps eBook - Addressing feedback

### LINKED ISSUE #31981: no more ConfigureServices
the "Configuring EF Core" text at https://learn.microsoft.com/en-us/dotnet/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps#configuring-ef-core makes explicit reference to ConfigureServices (i.e. the old Startup way of achieving ConfigureServices and Configure) whereas latest code condenses this  into the main code instead. Likewise the subsequent para again explicitly notes ConfigureServices [there are currently 5 refs to "ConfigureServices "]. Better if these docs are less [or more!] specific.

Incidentally the "a connection string defined in Configuration" text is also erroneous as that Configuration is not itself a string but a container and requires the GCS method call to lookup the stated DefaultConnection key to find the actual connectionstring. Again that old-style may have previously captured that container in a ctor but the docs here merely invent it as magic.

The simplistic "Eager=good, lazy=bad" guidance makes no mention of the "Cartesian Explosion" downside (cf. https://www.thinktecture.com/en/entity-framework-core/cartesian-explosion-problem-in-3-1/), nor the .AsSplitQuery() consideration. I accept this may be too deep for an intro doc, but it does mention the Dev vs Prod perf issues.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 8e1a443c-c3dd-b865-58b2-0ff0902308ce
* Version Independent ID: d974261f-2544-dee9-f401-e02d715012cb
* Content: [Work with data in ASP.NET Core Apps](https://learn.microsoft.com/en-us/dotnet/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps)
* Content Source: [docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md](https://github.com/dotnet/docs/blob/main/docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md)
* Product: **dotnet-architecture**
* Technology: **modern-web-apps-azure**
* GitHub Login: @ardalis
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by ardalis**
## Summary

Addressing feedback and updating for Program.cs

Fixes #31981

### CODE REVIEW

**Comment by ardalis** [comment]
@IEvangelist

File: docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md [hunk 1]
```diff
@@ -54,20 +54,20 @@ Your DbContext must have a constructor that accepts `DbContextOptions` and pass
 
 ### Configuring EF Core
 
-In your ASP.NET Core application, you'll typically configure EF Core in your `ConfigureServices` method. EF Core uses a `DbContextOptionsBuilder`, which supports several helpful extension methods to streamline its configuration. To configure CatalogContext to use a SQL Server database with a connection string defined in Configuration, you would add the following code to `ConfigureServices`:
+In your ASP.NET Core application, you'll typically configure EF Core in `Program.cs` with your application's other dependencies. EF Core uses a `DbContextOptionsBuilder`, which supports several helpful extension methods to streamline its configuration. To configure CatalogContext to use a SQL Server database with a connection string defined in Configuration, you would add the following code:
```
Comment (IEvangelist) [inline docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md:57]:
```suggestion
In your ASP.NET Core application, you'll typically configure EF Core in _Program.cs_ with your application's other dependencies. EF Core uses a `DbContextOptionsBuilder`, which supports several helpful extension methods to streamline its configuration. To configure CatalogContext to use a SQL Server database with a connection string defined in Configuration, you would add the following code:
```

File: docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md [hunk 2]
```diff
@@ -54,20 +54,20 @@ Your DbContext must have a constructor that accepts `DbContextOptions` and pass
...
 
 ```csharp
-services.AddDbContext<CatalogContext>(options => options.UseSqlServer (Configuration.GetConnectionString("DefaultConnection")));
+builder.Services.AddDbContext<CatalogContext>(options => options.UseSqlServer (builder.Configuration.GetConnectionString("DefaultConnection")));
```
Comment (IEvangelist) [inline docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md:60]:
```suggestion
builder.Services.AddDbContext<CatalogContext>(
    options => options.UseSqlServer(
        builder.Configuration.GetConnectionString("DefaultConnection")));
```

File: docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md [hunk 3]
```diff
@@ -148,6 +148,8 @@ _Lazy loading_ is a feature that automatically loads related data as it is refer
 
 [Avoid Lazy Loading Entities in Web Applications](https://ardalis.com/avoid-lazy-loading-entities-in-asp-net-applications)
 
+It's a good idea to test your application while examining the actual database queries it makes. Under certain circumstances, EF Core may make many more queries or a more expensive query than is optimal for the application. One such problem is known as a [Cartesian Explosion](https://learn.microsoft.com/en-us/ef/core/querying/single-split-queries#cartesian-explosion). The EF Core team makes available the [`AsSplitQuery` method](https://learn.microsoft.com/en-us/ef/core/querying/single-split-queries#split-queries) as one of several ways to tune runtime behavior.
```
Comment (IEvangelist) [inline docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md:151]:
```suggestion
It's a good idea to test your application while examining the actual database queries it makes. Under certain circumstances, EF Core may make many more queries or a more expensive query than is optimal for the application. One such problem is known as a [Cartesian Explosion](/ef/core/querying/single-split-queries#cartesian-explosion). The EF Core team makes available the [AsSplitQuery method](/ef/core/querying/single-split-queries#split-queries) as one of several ways to tune runtime behavior.
```

File: docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md [hunk 4]
```diff
@@ -465,7 +467,7 @@ public class CachedCatalogService : ICatalogService
 }
 ```
 
-To configure the application to use the cached version of the service, but still allow the service to get the instance of CatalogService it needs in its constructor, you would add the following lines in `ConfigureServices`:
+To configure the application to use the cached version of the service, but still allow the service to get the instance of CatalogService it needs in its constructor, you would add the following lines in `Program.cs`:
```
Comment (IEvangelist) [inline docs/architecture/modern-web-apps-azure/work-with-data-in-asp-net-core-apps.md:470]:
```suggestion
To configure the application to use the cached version of the service, but still allow the service to get the instance of CatalogService it needs in its constructor, you would add the following lines in _Program.cs_:
```

**Review by IEvangelist — CHANGES_REQUESTED** [review]
A few nits, otherwise looks good.

**Review by IEvangelist — APPROVED** [review]

**Comment by ardalis** [comment]
Looks good, thanks!

**Pull request merged**